core/src/test/java/org/jdbi/v3/core/mapper/MapEntryMapperTest.java

```diff
@@ -0,0 +1,178 @@
+        h.execute("create table config (key varchar, value varchar)");
+        h.prepareBatch("insert into config (key, value) values (?, ?)")
+                .add("foo", "123")
+                .add("bar", "xyz")
+                .execute();
+
+        Map<String, String> map = h.createQuery("select key, value from config")
+                .configure(MapEntryMapper.Config.class, cfg -> cfg.setKeyColumn("key").setValueColumn("value"))
+                .collectInto(new GenericType<Map<String, String>>() {});
+
+        assertThat(map).containsOnly(
+                entry("foo", "123"),
+                entry("bar", "xyz"));
+    }
+
+    @Test
+    public void uniqueIndex() {
+        h.execute("create table user (id int, name varchar)");
+        h.prepareBatch("insert into user (id, name) values (?, ?)")
+                .add(1, "alice")
+                .add(2, "bob")
+                .add(3, "cathy")
+                .add(4, "dilbert")
+                .execute();
+
+        Map<Integer, User> map = h.createQuery("select * from user")
+                .configure(MapEntryMapper.Config.class, cfg -> cfg.setKeyColumn("id"))
+                .registerRowMapper(ConstructorMapper.of(User.class))
+                .collectInto(new GenericType<Map<Integer, User>>() {});
+
+        assertThat(map).containsOnly(
+                entry(1, new User(1, "alice")),
+                entry(2, new User(2, "bob")),
+                entry(3, new User(3, "cathy")),
+                entry(4, new User(4, "dilbert")));
+    }
+
+    @Test
+    public void joinRow() {
+        h.execute("create table user (id int, name varchar)");
+        h.execute("create table phone (id int, user_id int, phone varchar)");
+        h.prepareBatch("insert into user (id, name) values (?, ?)")
+                .add(1, "alice")
+                .add(2, "bob")
+                .add(3, "cathy")
+                .execute();
+        h.prepareBatch("insert into phone (id, user_id, phone) values (?, ?, ?)")
+                .add(10, 1, "555-0001")
+                .add(20, 2, "555-0002")
+                .add(30, 3, "555-0003")
+                .execute();
+
+        String sql = "select u.id u_id, u.name u_name, p.id p_id, p.phone p_phone " +
+                "from user u left join phone p on u.id = p.user_id";
+        Map<User, Phone> map = h.createQuery(sql)
+                .registerRowMapper(ConstructorMapper.of(User.class, "u"))
+                .registerRowMapper(ConstructorMapper.of(Phone.class, "p"))
+                .collectInto(new GenericType<Map<User, Phone>>() {});
+
+        assertThat(map).containsOnly(
+                entry(new User(1, "alice"), new Phone(10, "555-0001")),
+                entry(new User(2, "bob"),   new Phone(20, "555-0002")),
+                entry(new User(3, "cathy"), new Phone(30, "555-0003")));
+    }
+
+    @SuppressWarnings("unchecked")
+    private <K, V, T> Collector<Map.Entry<K, V>, ?, T> intoContainer(GenericType<T> containerType) {
```

Comment:
Good catch. This was used previously, but isn't needed now we have `ResultBearing.collectInfo(GenericType)`